### ISSUE TITLE
Merge conflicts for concurrent release PRs in monorepo setting

### ISSUE THREAD

**Issue by strieflin** [googleapis/release-please#1870]
Hi all,

we are evaluating release-please with the manifest approach in a monorepo setting. If we have release PRs open for two or more packages / components in parallel, merging one of them creates a merge conflict for the others because of conflicting changes in the `.release-please-manifest.json` file. What is the recommended workflow to deal with this situation?

Thanks a lot for any comments or pointers to existing repos or documentation in advances.

**Comment by sswastioyono18** [comment]
actually just realized this a week ago when I want to move into monorepo as well.

I was thinking whether it is possible to make each package/module their own `.release-please-manifest.json` and the one in the root of monorepo will just refer to each module/package? 

it's still possible to have conflict but less often especially if it is just regular release (increase version).

OR maybe move it together into the `release-please-config.json` ? after all, it's already structured by package

**Comment by philippeboyd** [comment]
+1 currently in the same boat... If only the resulted json in `.release-please-manifest.json` was NOT minified, it would already help a lot? Not sure what's the logic behind minifying the content of `.release-please-manifest.json`

**Comment by alex-enchi** [comment]
Have you tried to just prettify `.release-please-manifest.json`? It wont be minified on subsequent runs.

**Comment by StefanBrand** [comment]
Is this a duplicate of #1502?

**Comment by shinebayar-g** [comment]
> Have you tried to just prettify .release-please-manifest.json? It wont be minified on subsequent runs.

Unfortunately it doesn't work when the updated modules are next to each other.

**Comment by shinebayar-g** [comment]
I got a hacky workaround working. I noticed that release-please actually rebases all the PRs that doesn't have `autorelease: pending` label attached. So if I remove that label from PRs and re-run release-please once more, then release-please fixes the conflicted PRs.

I tried to check for "CONFLICTING" status specifically to reduce the noise, but unfortunately mergeable status doesn't become CONFLICTING until I browse the Pull Request page. Then it becomes CONFLICTING.

```yaml
on:
  push:
    branches:
      - main

permissions:
  contents: write
  pull-requests: write
  repository-projects: read # needed for 'gh pr edit' https://github.com/cli/cli/issues/6274

name: release-please

jobs:
  release-please:
    runs-on: ubuntu-22.04
    outputs:
      releases_created: ${{ steps.release.outputs.releases_created }}
    steps:
      - uses: google-github-actions/release-please-action@v3
        id: release
        with:
          command: manifest

  check-conflicting-prs:
    runs-on: ubuntu-22.04
    needs: release-please
    if: needs.release-please.outputs.releases_created == 'true'
    outputs:
      need_rebase: ${{ steps.get-pending-prs.outputs.need_rebase }}
    steps:
      - name: Get pending PRs
        id: get-pending-prs
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: |
          pending_prs=$(gh pr list --repo ${{ github.repository}} --label "autorelease: pending" --state open --json number --jq '.[].number')
          conflicted_prs=""
          if [[ -n "$pending_prs" ]]; then
              for pr_num in $pending_prs; do
                  echo "Checking pr status: $pr_num"
                  mergeable=$(gh pr view --repo ${{ github.repository}} "$pr_num" --json mergeable --jq '.mergeable')
                  echo "mergeable: $mergeable"
                  if [[ "$mergeable" == "CONFLICTING" ]]; then
                      echo "pr: $pr_num has a conflict."
                      conflicted_prs+=" $pr_num"
                  elif [[ "$mergeable" == "UNKNOWN" ]]; then
                      echo "pr: $pr_num has an UNKNOWN status."
                      conflicted_prs+=" $pr_num"
                  fi
              done
          else
              echo "No pending PRs found."
              exit 0
          fi

          if [[ -n "$conflicted_prs" ]]; then
              echo "CONFLICTING or UNKNOWN status PRs found."
              for pr_num in $conflicted_prs; do
                  echo "removing 'autorelease: pending' label from pr: $pr_num"
                  gh pr edit --repo ${{ github.repository}} "$pr_num" --remove-label "autorelease: pending"
              done
              echo "need_rebase=true"
              echo "need_rebase=true" >> "$GITHUB_OUTPUT"
          else
              echo "No conflicted PRs found."
          fi

  release-please-rebase:
    needs: check-conflicting-prs
    runs-on: ubuntu-22.04
    if: needs.check-conflicting-prs.outputs.need_rebase == 'true'
    steps:
      - uses: google-github-actions/release-please-action@v3
        id: release
        with:
          command: manifest

```

**Comment by michaeldev5** [comment]
https://github.com/googleapis/release-please/issues/1870#issuecomment-1748390833

**Comment by kolja-ec** [comment]
If the solution to this problem is: Remove autorelease: pending tag from the PRs with merge-conflicts? Why not always remove this tag upfront in a release-please run in general to fix this problem?

**Comment by Kerwood** [comment]
This is still an issue. Any progress on this ?

**Comment by Kerwood** [comment]
Not adding the `autorelease: pending` does not make release-please rebase the branch. I added a custom label using the `--label` and the conflict issue persists.
